cylc/flow/install.py

```diff
@@ -285,13 +285,15 @@ def install_workflow(
     source = Path(expand_path(source)).resolve()
     if not workflow_name:
         workflow_name = get_source_workflow_name(source)
-    validate_workflow_name(workflow_name, check_reserved_names=True)
     if run_name is not None:
+        validate_workflow_name(os.path.join(workflow_name, run_name))
         if len(Path(run_name).parts) != 1:
             raise WorkflowFilesError(
                 f'Run name cannot be a path. (You used {run_name})'
             )
         check_reserved_dir_names(run_name)
```

Comment:
Still need to check reserved dir names on the whole path
```suggestion
        validate_workflow_name(
            os.path.join(workflow_name, run_name),
            check_reserved_names=True
        )
        if len(Path(run_name).parts) != 1:
            raise WorkflowFilesError(
                f'Run name cannot be a path. (You used {run_name})'
            )
```

Reply:
@wxtim I think there ought to be a test that covers this point

Reply:
I think there ought to be a test that covers this point